box/src/main/scala/blobstore/box/BoxStore.scala

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2018 LendUp Global, Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package blobstore
+package box
+
+
+import java.io.{InputStream, OutputStream, PipedInputStream, PipedOutputStream}
+
+import cats.effect.Effect
+import com.box.sdk.{BoxAPIConnection, BoxFile, BoxFolder, BoxItem}
+import fs2.{Sink, Stream}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+case class BoxStore[F[_]](api: BoxAPIConnection, rootFolderId: String)(implicit F: Effect[F], ec: ExecutionContext) extends Store[F] {
+
+  val rootFolder = new BoxFolder(api, rootFolderId)
+
+  def boxItemAtPath(path: Path): Option[BoxItem] = {
+    boxItemAtPath(rootFolder, path.root :: path.key.split("/").toList)
+  }
+
+  def boxItemAtPath(parentFolder: BoxFolder, pathParts: List[String]): Option[BoxItem] = {
+    pathParts match {
+      case Nil => None
+      case head::Nil =>
+        parentFolder.getChildren.asScala.find(info => info.getName.equalsIgnoreCase(head))
+          .map(_.getResource.asInstanceOf[BoxItem])
+      case head::tail =>
+        parentFolder.getChildren.asScala.find(info => info.getName.equalsIgnoreCase(head))
+          .flatMap(info => boxItemAtPath(new BoxFolder(api, info.getID), tail))
+    }
+  }
+
+  def boxFileAtPath(path: Path): Option[BoxFile] = {
+    boxItemAtPath(path).flatMap {
+      case file: BoxFile => Some(file)
+      case _ => None
+    }
+  }
+
+  def boxFolderAtPath(path: Path): Option[BoxFolder] = {
+    boxItemAtPath(path).flatMap {
+      case folder: BoxFolder => Some(folder)
+      case _ => None
+    }
+  }
+
+  /**
+    * List paths. See [[StoreOps.ListOps]] for convenient listAll method.
+    *
+    * @param path to list
+    * @return stream of Paths. Implementing stores must guarantee that returned Paths
+    *         have correct values for size, isDir and lastModified.
+    */
+  override def list(path: Path): fs2.Stream[F, Path] = {
+    for {
+      itemOpt <- Stream.eval(F.delay(boxItemAtPath(path)))
+      item <- if (itemOpt.isDefined) Stream.emit(itemOpt.get) else Stream.empty
```

Comment:
I'd prefer we never call Option#get, and I made a PR with one suggested alternative. another alternative 
`itemOpt.fold[Stream[F,BoxItem]](Stream.empty)(Stream.emit(_))`